version: '3'

services:
  web:
    container_name: Django_App
    build: .
    command: ["sh", "./start.sh"]
    volumes:
      - .:/app
      - ./images:/app/images
    env_file:
      - .env
    ports:
      - "8000:8000"
    depends_on:
      - db

  db:
    image: postgres:16
    container_name: Postgres_DB
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: Chaotix_ai_Assignment_DB
      POSTGRES_USER: dee777
      POSTGRES_PASSWORD: 1234
    ports:
    - "5432:5432"

  pgadmin:
    image: dpage/pgadmin4:latest
    container_name: Pg_Admin
    ports:
      - "5050:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: test1@email.com
      PGADMIN_DEFAULT_PASSWORD: test1
    volumes:
      - pgadmin_data:/var/lib/pgadmin
    depends_on:
      - db

  redis:
    container_name: Redis
    image: redis:6
    ports:
      - "6379:6379"

  celery:
    container_name: Celery
    build: .
    command: celery -A Chaotix_ai_Assignment worker --loglevel=info
    env_file:
      - .env
    volumes:
      - .:/app
      - ./images:/app/images
    depends_on:
      - redis
      - web

volumes:
  postgres_data:
  pgadmin_data: